In [1]:
import pandas as pd
import numpy as np
import scipy as sp
import matplotlib
import matplotlib.pyplot as plt
import matplotlib.cm as cmx
import matplotlib.colors as colors
from mpl_toolkits.mplot3d import Axes3D
%matplotlib inline

In [2]:
national_names = pd.read_csv('Datasets/NationalNames.csv', delimiter = ',', usecols = [1, 2, 3, 4])
state_names = pd.read_csv('Datasets/StateNames.csv', delimiter = ',', usecols = [1, 2, 3, 4, 5])

In [3]:
national_names.head()

Name  Year Gender  Count
0       Mary  1880      F   7065
1       Anna  1880      F   2604
2       Emma  1880      F   2003
3  Elizabeth  1880      F   1939
4     Minnie  1880      F   1746

In [4]:
state_names.head()

Name  Year Gender State  Count
0      Mary  1910      F    AK     14
1     Annie  1910      F    AK     12
2      Anna  1910      F    AK     10
3  Margaret  1910      F    AK      8
4     Helen  1910      F    AK      7

### Question 1: Can you identify any limitations or distortions of the data?

The years included in the national data set and the state data set do not match up. The national data begins in 1880 while the state data begins in 1910. This limits the comparisons we can make between state and national data.

We not have totals for each name in the data set since the data is broken up by year. This means we have to calculate those separately by summing up the counts of each name by year.

There is a different number of names for each row and for each state because the criteria used for including the name in the data set is that the name has to have a minimum of five occurrences. Thus we cannot assume that the number of names in the data set will be the same for any two years.

The data is not broken up into smaller components that could make analysis easier. When analyzing the data, we have to look at the data frames in their entirety, which will be difficult since the data frames are so large.